docs/iris/src/whatsnew/make.py

```diff
@@ -0,0 +1,120 @@
+## (C) British Crown Copyright 2015, Met Office
+#
+# This file is part of Iris.
+#
+# Iris is free software: you can redistribute it and/or modify it under
+# the terms of the GNU Lesser General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Iris is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with Iris. If not, see <http://www.gnu.org/licenses/>.
+
+import datetime
+import os
+import sys
+import re
+from operator import itemgetter
+from distutils import version
+
+SOFTWARE_NAME = "Iris"
+VALID_CATEGORIES = [
+    {"Prefix": "bugfix", "Title": "Bugs Fixed"},
+    {"Prefix": "newfeature", "Title": "Features"},
+    {"Prefix": "incompatiblechange", "Title": "Incompatible Changes"},
+    {"Prefix": "deprecate", "Title": "Deprecations"},
+    {"Prefix": "docchange", "Title": "Documentation Changes"}
+]
+
+def get_release_number(directory_name):
+    """ Takes a formatted string and returns a LooseVersion object. """
+    regular_expression = "(.*)\\_(.*)"
```

Comment:
when using regex in functions, it is often recommended to compile the regex once, then use it many times